dropbox/dropbox_client.py

```diff
@@ -529,10 +549,24 @@ def request_json_string(self,
         url = self._get_route_url(fq_hostname, func_name)
 
         headers = {'User-Agent': self._user_agent}
-        if host != HOST_NOTIFY:
+        if auth_type == USER_AUTH or auth_type == TEAM_AUTH:
             headers['Authorization'] = 'Bearer %s' % self._oauth2_access_token
             if self._headers:
                 headers.update(self._headers)
+        elif auth_type == APP_AUTH:
+            if self._app_key is None or self._app_secret is None:
+                raise BadInputException('A client id and secret is required for this function')
```

Comment:
this 'function' might not be clear wording what is happening. Maybe a message like 

'Client Id and Client Secret are required for routes with App auth'?

Reply:
Actually now I think we don't really need this check here. The constructor of the class already checks that both app key and app secret are provided. But to be extra cautious, we can make it an assertion to be sure that nothing has changed between class initiation and the app auth API call